controllers/machinehealthcheck_controller.go

```diff
@@ -356,25 +353,21 @@ func (r *MachineHealthCheckReconciler) createClusterNodeInformer(ctx context.Con
 		return errors.Wrap(err, "error fetching remote cluster config")
 	}
 
-	k8sClient, err := kubernetes.NewForConfig(config)
+	clusterCache, err := cache.New(config, cache.Options{})
 	if err != nil {
-		return errors.Wrap(err, "error constructing remote cluster client")
+		return errors.Wrap(err, "error creating cache for remote cluster")
 	}
-
-	// TODO(JoelSpeed): See if we use the resync period from the manager instead of 0
-	factory := informers.NewSharedInformerFactory(k8sClient, 0)
-	nodeInformer := factory.Core().V1().Nodes().Informer()
-	go nodeInformer.Run(ctx.Done())
+	go clusterCache.Start(ctx.Done())
```

Comment:
xref #2577: I think we're going to need to able to stop a cache when the associated Cluster is deleted. This probably means creating a stop channel per cache and tracking it in the map, and adding logic that closes the stop channel for a cache if its corresponding Cluster is not found when reconciling.